File: back/src/main/java/com/tdevilleduc/urthehero/back/service/IPersonService.java
package com.tdevilleduc.urthehero.back.service;

import com.tdevilleduc.urthehero.back.model.Person;
import com.tdevilleduc.urthehero.back.model.PersonDTO;

import java.util.List;
import java.util.Optional;

public interface IPersonService {

    boolean exists(Integer personId);
    boolean notExists(Integer personId);
    Optional<Person> findById(Integer personId);
    List<Person> findAll();
    PersonDTO createOrUpdate(PersonDTO personDto);
    void delete(Integer personId);
}

File: back/src/main/java/com/tdevilleduc/urthehero/back/service/IPersonService.kt
package com.tdevilleduc.urthehero.back.service

import com.tdevilleduc.urthehero.back.model.Person
import com.tdevilleduc.urthehero.back.model.PersonDTO

interface IPersonService {
    fun exists(personId: Int?): Boolean
    fun notExists(personId: Int?): Boolean
    fun findById(personId: Int): Person
    fun findByLogin(userName: String): Person
    fun findAll(): MutableList<Person>
    fun createOrUpdate(personDto: PersonDTO): PersonDTO
    fun delete(personId: Int)
}